app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -160,6 +166,18 @@ class WireActivityViewModel @Inject constructor(
         }
     }
 
+    fun observerE2EIState() {
+        viewModelScope.launch(dispatchers.io()) {
+            observeUserId
+                .flatMapLatest {
+                    it?.let { observeIfE2EIRequiredDuringLoginUseCaseProviderFactory.create(it).observeIfE2EIIsRequiredDuringLogin() }
+                        ?: flowOf(null)
+                }
+                .distinctUntilChanged()
+                .collect { _observeE2EIState.emit(it) }
+        }
+    }
+
```

Comment:
instead of having 2 `val`s `_observeE2EIState` and `observeE2EIState` could be just one:
```
private val observeE2EIState = observeUserId
    .flatMapLatest {
        it?.let { observeIfE2EIRequiredDuringLoginUseCaseProviderFactory.create(it).observeIfE2EIIsRequiredDuringLogin() }
            ?: flowOf(null)
    }
    .distinctUntilChanged()
```

Reply:
@borichellow thanks, let's have this change in another pr to merge it later, to not disturb this PR.